script.js

```diff
@@ -0,0 +1,193 @@
+
+// * Create your form (html / css)
+// * Prevent default and show an alert on form submit
+// * Use AJAX to make a request. Show data in console
+// * Create an array of image URLs (tip: use [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) and [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)).
+// * Make the form / title / description hide
+// * Cycle through images
+//     * tip: use [setInterval](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setInterval)
+//     * Maybe just set the src attribute of a single image tag for now
+// * Add some interesting style / animation
+// * Create button to stop animation (tip: use [clearInterval](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/clearInterval)).
+var imageArray = [];
+var counter = 0;
+var timer;
+$('#stop-slides').hide();
+
+$('#myForm').submit(function (event) {
+  event.preventDefault();
+  $('#photo-div').html("<p>Loading...</p>");
+  var input = $('#query');
+  var userText = input.val();
+var searchStr = userText + ' nsfw:no site:i.imgur.com';
+ 
+   
+    console.log('Starting search for ' + searchStr);
+    $.get('https://www.reddit.com/search.json', {q: searchStr, limit: 100}).done(function(response){
+    console.log(response);
+
+    // var results = response.data.children
+    // var redditData = results[i].data;
+
+    response.data.children.filter(function (result) {
+      if (result.data.preview && result.data.preview.images[0].source.url) {
+         imageArray.push(result.data.preview.images[0].source.url);
+        console.log(imageArray);
+        //return true;
+      }
+ 
+      
+ 
+     });
+  triggerTimer();
+  $('#query').val("");
+  });
+
+  });
+ 
+
+
+
+  //   var results = response.data.children
+  //   for (i = 0; i < results.length; i++) {
+  //     var redditData = results[i].data;
+  //     if (redditData.domain === "i.imgur.com" && redditData.url) {
+  //       imageArray.push(redditData.url);
+  //     }
+  //     console.log(imageArray);
+
+
+
+function clearSearchResults() {
+  $("#results").html("");
+  $('#photo-div').html(""); 
+  $('#myForm').fadeIn(1000);
+
+  imageArray = [];
+  
+}
+
+$('#submit')
+  .click(function () {
+    // $('#myForm').hide();
+    $('#myForm').fadeOut(500);
+    $('#stop-slides').fadeIn(500);
+});
+
+$('#stop-slides').click(function() {
+  console.log('stop click');
+  stop();
+  $('#stop-slides').hide();
+  clearSearchResults();
+});
+
+
+
+function flashImage() {
+    if ($('#photo-div').html()) {
+      $('#photo-div').html("");
+    }
+    var html = '<img class="image-responsive img-thumbnail" src="' + imageArray[counter] + '"/>'; 
+    $('#photo-div').append(html); 
+    if (counter >= (imageArray.length-1)) {
+      stop();
+      clearSearchResults();
```

Comment:
because clearSearchResults is called in the stop function, you don't have to call it here as well, slight redundancy.